src/coreclr/jit/valuenum.cpp

```diff
@@ -7793,10 +7793,44 @@ ValueNum ValueNumStore::EvalHWIntrinsicFunBinary(var_types      type,
             }
 
 #ifdef TARGET_ARM64
-            case NI_AdvSimd_Multiply:
             case NI_AdvSimd_MultiplyByScalar:
-            case NI_AdvSimd_Arm64_Multiply:
             case NI_AdvSimd_Arm64_MultiplyByScalar:
+            {
+                if (!varTypeIsFloating(baseType))
+                {
+                    // Handle `x * 0 == 0` and `0 * x == 0`
+                    // Not safe for floating-point when x == -0.0, NaN, +Inf, -Inf
+                    ValueNum zeroVN = VNZeroForType(TypeOfVN(cnsVN));
+
+                    if (cnsVN == zeroVN)
+                    {
+                        return VNZeroForType(type);
+                    }
+                }
+
+                // Handle x * 1 => x, but only if the scalar RHS is 1.
+                ValueNum oneVN;
+                if (varTypeIsSIMD(TypeOfVN(arg1VN)))
+                {
+                    oneVN = VNOneForSimdType(TypeOfVN(arg1VN), baseType);
+                }
+                else
+                {
+                    oneVN = VNOneForType(baseType);
+                }
```

Comment:
Can this actually be hit given that we expect `TYP_SIMD8` for the second operand?

Reply:
Good question... let me add some strong asserts and simplify this a bit.

Reply:
I think the same path down in the regular multiply handling is also "dead". It was likely intended to handle the `vector * scalar` case, but we don't actually ever produce a `GT_HWINTRINSIC(TYP_SIMD, TYP_SCALAR)` node anymore, afaict.

We appear to always normalize to `Broadcast` on x86/x64. We similarly normalize to `Broadcast` on `Arm64` except for when a `MultiplyByScalar` variant exists, in which case we always insert `CreateScalarUnsafe`.

I then don't see any overloads of the instructions that would take a proper scalar register, so I don't think we'd ever hit it in practice.